client.go

```diff
@@ -17,6 +17,7 @@ var ProcessExpiredAdInterval = 1 * time.Minute
 
 type CacheFunc func(string, time.Duration) (string, error)
 type EventFunc func(string, string, string, string)
+type SaveFailedPOP func(*PoPRequest)
```

Comment:
Naming: FailedPoPFunc or FailedPoPHandler might be better as it's more generic. In this package, it's not really our concern what happens w/ failed pops.

Reply:
done